test/integration/gardenlet/seed/seed/seed_test.go

```diff
@@ -196,12 +215,27 @@ var _ = Describe("Seed controller tests", func() {
 					}
 					Expect(testClient.Create(ctx, globalMonitoringSecret)).To(Succeed())
 
+					By("Wait until the manager cache observes the global monitoring secret")
+					Eventually(func() error {
+						return mgrClient.Get(ctx, client.ObjectKeyFromObject(globalMonitoringSecret), globalMonitoringSecret)
+					}).Should(Succeed())
+
 					DeferCleanup(func() {
 						Expect(testClient.Delete(ctx, globalMonitoringSecret)).To(Succeed())
 					})
 				})
 
 				test := func(seedIsGarden bool) {
+					// Make sure there are no garden resources present before proceeding to the test, since the controller takes the
+					// decision whether the seed is garden or not based on this.
+					if !seedIsGarden {
+						Eventually(func(g Gomega) []operatorv1alpha1.Garden {
+							gardenList := &operatorv1alpha1.GardenList{}
+							g.Expect(mgrClient.List(ctx, gardenList)).To(Succeed())
+							return gardenList.Items
+						}).Should(BeEmpty())
+					}
+
```

Comment:
Do we still need this after the recent changes?

Reply:
Hmm, this might not be useful, but we can still wait in the `garden` test `BeforeEach` for the garden resource to be gone. Will make the changes.